backend/internal/service/handler/auth.go

```diff
@@ -0,0 +1,155 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"platnm/internal/errs"
+	"sync"
+	"time"
+
+	"net/http"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/google/uuid"
+	"golang.org/x/oauth2"
+)
+
+var (
+	SpotifyAuthURL  = "https://accounts.spotify.com/authorize"
+	SpotifyTokenURL = "https://accounts.spotify.com/api/token"
+)
+
+type TokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	Scope        string `json:"scope"`
+	ExpiresIn    int    `json:"expires_in"`
+	RefreshToken string `json:"refresh_token"`
+}
+
+// this is used to keep track of associated code verifier for each state since code verifier will be needed again to get token
+type CodeVerifierStore struct {
```

Comment:
just wrap [fiber's memory storage](https://github.com/gofiber/storage/tree/main/memory)

Reply:
also, does this need to be exported?

Reply:
if it uses the interface value of `storage.Storage`, you can swap out for memcache or redis if you want down the line

Reply:
yes, you can. it defaults to using cookies which @alderwhiteford and I struggled with on SAC. here is how i got it working with headers -> https://github.com/GenerateNU/sac/blob/main/backend/integrations/oauth/soth/sothic/sothic.go